src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp

```diff
@@ -345,9 +338,11 @@ void ReplicatedMergeTreeRestartingThread::partialShutdown(bool part_of_full_shut
 
     storage.partial_shutdown_called = true;
     storage.partial_shutdown_event.set();
-    storage.replica_is_active_node = nullptr;
+    storage.replica_is_active_node.reset();
```

Comment:
If we assume here that `replica_is_active_node.reset()` always have to lead to calling its destructor, then `replica_is_active` should be `unique_ptr`, not `shared_ptr`

Reply:
`replica_is_active_node` seems to be stored only inside storage, but better to return back `= nullptr`